docs/source/userguide/newton_solver.rst

```diff
@@ -0,0 +1,90 @@
+
+.. dropdown:: Linear solver methods
+
+    * "bicgstab" - Biconjugate gradient stabilized method
+    * "cg" - Conjugate gradient method
+    * "gmres" - Generalized minimal residual method
+    * "minres" - Minimal residual method
+    * "mumps" - MUMPS (MUltifrontal Massively Parallel Sparse direct Solver)
+    * "petsc" - PETSc built in LU solver
+    * "richardson" - Richardson method 
+    * "superlu" - SuperLU
+    * "superlu_dist" - Parallel SuperLU
+    * "tfqmr" - Transpose-free quasi-minimal residual method
+    * "umfpack" - UMFPACK (Unsymmetric MultiFrontal sparse LU factorization)
+
+The preconditioner can be set with the ``preconditioner`` attribute. The list of available preconditioners can be viewed with: ``print(fenics.list_krylov_solver_preconditioners())``.
+
+.. dropdown:: Preconditioners
+
+    * "amg" - Algebraic multigrid
+    * "hypre_amg" - Hypre algebraic multigrid (BoomerAMG)
+    * "hypre_euclid" - Hypre parallel incomplete LU factorization
+    * "hypre_parasails" - Hypre parallel sparse approximate inverse
+    * "icc" - Incomplete Cholesky factorization
+    * "ilu" - Incomplete LU factorization
+    * "jacobi" - Jacobi iteration 
+    * "petsc_amg" - PETSc algebraic multigrid
+    * "sor" - Successive over-relaxation
+
+Similarly, the Newton solver parameters of :class:`festim.HeatTransferProblem`, :class:`festim.ExtrinsicTrap`, or :class:`festim.NeutronInducedTrap` 
+can be defined upon the need. Here is an example for the heat transfer problem:
+
+.. code-block:: python
+
+    model.T = HeatTransferProblem(
+        transient=True,
+        initial_condition=300,
+        absolute_tolerance=1.0,
+        relative_tolerance=1e-10,
+        maximum_iterations=50,
+        linear_solver="gmres",
+        preconditioner="icc")
+
+
+--------------
+Custom solver
+--------------
+
+For a more flexible control, the built-in Newton solver can be overwritten by a custom solver based on the ``fenics.NewtonSolver`` class. 
+A user-defined Newton solver can be provided after :class:`festim.Simulation.initialise()`:
+
+.. code-block:: python
+
+    class CustomSolver(f.NewtonSolver):
+        def __init__(self):
+            super().__init__()
+
+    custom_solver = CustomSolver()
+    custom_solver.parameters["error_on_nonconvergence"] = False
+    custom_solver.parameters["absolute_tolerance"] = 1e10
+    custom_solver.parameters["relative_tolerance"] = 1e-10
+    custom_solver.parameters["maximum_iterations"] = 100
+    custom_solver.parameters["linear_solver"] = "gmres"
+    custom_solver.parameters["preconditioner"] = "ilu"
+
+    model.initialise()
+
+    model.h_transport_problem.newton_solver = custom_solver
+
+    model.run()
+
+.. warning::
+    
+    For a stationary heat transfer problem, a custom Newton solver has to be provided before the simulation initialisation! 
```

Comment:
Could we add another warning saying that this will override the solver parameters given in `Settings`?